test/Microsoft.ML.Functional.Tests/ModelLoading.cs

```diff
@@ -74,35 +81,49 @@ public void LoadModelAndExtractPredictor()
             {
                 // This model can be loaded either as a composite data loader,
                 // a transformer model + an input schema, or a transformer model + a data loader.
-                var t = _ml.Model.LoadWithDataLoader(fs, out IDataLoader<IMultiStreamSource> l);
+                var t = _ml.Model.LoadWithDataLoader(fs, out loadedCompositeLoader);
+                // This is a bit strange, as it seems to test that it can reload from the same
+                // stream twice opened only once, which as far as I know is not really a requirement
```

Comment:
It also tests that you can use the same file to load using both `LoadWithDataLoader` and `Load` (which loads with the schema). Maybe we should open a separate stream for that then, to make sure that if it fails we know that it's a real problem and not just a problem with reading twice from the same stream?

Reply:
By the way, we already have overloads for loading that take file names, don't we?

---
In reply to: [267891929](https://github.com/dotnet/machinelearning/pull/3044#discussion_r267891929) [](ancestors = 267891929)

Reply:
Sure! I changed some tests to use those so both paths are tested, but this seems like a fine place to use them.

Reply:
Um, sorry, nevermind -- this is a little annoying dealing with all the local variables and changing of scope. Let's just leave it for now...